src/Workleap.DomainEventPropagation.Analyzers/EventDomainAttributeUsageAnalyzer.cs

```diff
@@ -61,13 +72,27 @@ public void AnalyzeOperationInvocation(SymbolAnalysisContext context)
             {
                 if (this.ImplementsBaseDomainEventInterface(classTypeSymbol))
                 {
-                    var hasDomainEventAttribute = classTypeSymbol.GetAttributes()
-                        .Any(x => x.AttributeClass != null && SymbolEqualityComparer.Default.Equals(x.AttributeClass, this._domainEventAttributeType));
+                    var domainEventAttribute = classTypeSymbol.GetAttributes()
+                        .FirstOrDefault(x => x.AttributeClass != null && SymbolEqualityComparer.Default.Equals(x.AttributeClass, this._domainEventAttributeType));
 
-                    if (!hasDomainEventAttribute)
+                    if (domainEventAttribute is null)
                     {
                         context.ReportDiagnostic(UseDomainEventAttribute, classTypeSymbol);
                     }
+
+                    if (domainEventAttribute is not null && domainEventAttribute.ConstructorArguments.Length == 1)
```

Comment:
It's best if C# roslyn analyzers are fully optimized, so here we can do `else if (ctor args length == 1)`, skipping a second null check